File: String/ReverseString.cpp
#include <iostream>
#include <string>
using namespace std;

int main(){
    string inputString;
    cin >> inputString;
    
    cout << "Original String: " << inputString;
    reverse(inputString.begin(), inputString.end());
    cout << "Reverse String: " << inputString;
    
    // Two pointer approach to get the original String
    int left = 0;
    int right = inputString.size() - 1;
    
    while(left < right){
        swap(inputString[left], inputString[right]);
        left++;
        right--;
    }
    
    cout << "Original Recovered String: " << inputString;
    
    return 0;
}

File: String/Java/ReverseString.java
java.util.*;

class ReverseString {
    public static void main(String args[]){
        Scanner input = new Scanner(System.in);
        String inputString = input.nextLine();
        
        System.out.println("Original String: " + inputString);
        char[] chars = inputString.toCharArray();
        
        int left = 0;
        int right = chars.length - 1;
        while(left < right){
            char temp = chars[left];
            chars[left] = chars[right];
            chars[right] = temp;
            left++;
            right--;
        }
        
        String reversed = new String(chars);
        System.out.println("Reversed String: " + reversed);
    }
}